cfme/tests/infrastructure/test_host.py

```diff
@@ -162,3 +180,22 @@ def test_tag_host_after_provider_delete(provider, appliance, setup_provider, req
     provider.wait_for_delete()
     tag = host.add_tag()
     request.addfinalizer(lambda: host.remove_tag(tag))
+
+
+def test_250_vmware_hosts_loading(appliance, create_250_hosts):
+    """
+    Test to automate BZ1580569
+    """
+    # Without the patch, this will cause the process to consume roughly 10+ Gigs of RAM
+    # due to a poorly optimized database query
+    rails_console = appliance.ssh_client.run_rails_console(
+        "MiqReport.load_from_view_options(Host, User.where(:userid => 'admin').first)"
+        ".paged_view_search", timeout=60
+    )
+    assert rails_console.success
+
+    # Check it loads in the UI fast
+    host_collection = appliance.collections.hosts
+    view = host_collection.appliance.browser.create_view(HostsView)
+    load_view = wait_for(lambda: view.entities.paginator.set_items_per_page(1000))
```

Comment:
I'm curious about the behavior that you expect here, `set_items_per_page` is implemented with `ReportDataControllerMixin`, which uses javascript to make this page change.  I imagine that JS command to change the per-page count is returning before the page is actually done rendering the entities.

I think you should consider making a query against `entities.get_first_entity` on the page in order to check that the entities are actually rendered, or just a wait that `entities.table.is_displayed` in a list view.

Consider parametrizing the test against tile, quad, and list views to cover all of those cases of capturing how long it took for the entities to be visible.

Reply:
thanks, Mike!